src/contrib/persistence/Akka.Persistence.Sql.Common/Snapshot/SqlSnapshotStore.cs

```diff
@@ -105,13 +105,20 @@ protected override void PostStop()
             _pendingRequestsCancellation.Cancel();
         }
 
-        private async Task<Initialized> Initialize()
+        private async Task<object> Initialize()
         {
-            using (var connection = CreateDbConnection())
+            try
             {
-                await connection.OpenAsync(_pendingRequestsCancellation.Token);
-                await QueryExecutor.CreateTableAsync(connection, _pendingRequestsCancellation.Token);
-                return Initialized.Instance;
+                using (var connection = CreateDbConnection())
+                {
+                    await connection.OpenAsync(_pendingRequestsCancellation.Token);
+                    await QueryExecutor.CreateTableAsync(connection, _pendingRequestsCancellation.Token);
+                    return Initialized.Instance;
+                }
+            }
+            catch (Exception e)
+            {
+                return new Failure {Exception = e};
```

Comment:
Similar to the change made to `SqlJournal`, this adds handling of exceptions during `Initialize` and passing a `Failure` message to `WaitForInitialization` so it can be handled and logged.